server/storage/endpoint/external_timestamp.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endpoint
+
+import (
+	"strconv"
+
+	"github.com/tikv/pd/pkg/errs"
+)
+
+// ExternalTimestamp is the external timestamp.
+// NOTE: This type is exported by HTTP API. Please pay more attention when modifying it.
+type ExternalTimestamp struct {
+	ExternalTimestamp uint64 `json:"external_timestamp"`
+}
+
+// ExternalTSStorage defines the storage operations on the external timestamp.
+type ExternalTSStorage interface {
+	LoadExternalTS() (uint64, error)
+	SaveExternalTS(timestamp uint64) error
+}
+
+var _ ExternalTSStorage = (*StorageEndpoint)(nil)
+
+// LoadExternalTS loads the external timestamp from storage.
+func (se *StorageEndpoint) LoadExternalTS() (uint64, error) {
+	value, err := se.Load(ExternalTimestampPath())
+	if err != nil || value == "" {
+		return 0, err
+	}
+	timestamp, err := strconv.ParseUint(value, 16, 64)
```

Comment:
why use 16bit?

Reply:
I found that gc-safe-point and min-resolved-ts are saved directly with uint64, but tso is converted to bytes for saving. Does external-ts need to be in the same format as tso, or any one of them? cc @JmPotato

Reply:
external-ts set from cdc, so it should consist with tso,

Reply:
Do they also need to be stored in the same format in etcd?